src/stream/src/common/table/state_table.rs

```diff
@@ -908,9 +908,15 @@ where
         // Tick the watermark buffer here because state table is expected to be committed once
         // per epoch.
         self.watermark_buffer_strategy.tick();
-        self.seal_current_epoch(new_epoch.curr)
-            .instrument(tracing::info_span!("state_table_commit"))
-            .await?;
+        if !self.is_dirty() {
+            // If the state table is not modified, go fast path.
+            self.local_store.seal_current_epoch(new_epoch.curr);
```

Comment:
How fast can this path be? 😄 I'm just worrying that someone modifies `StateTable::seal_current_epoch` but forget to change this.

Reply:
This path is just a replacement for `commit_no_data_expected`, later we may remove the `commit_no_data_expected` method. I just checked the code and think we should check `is_dirty` in local store `flush` instead of in state table.